### ISSUE TITLE
Failed to get VPN serves data

### ISSUE THREAD

**Issue by wuhualu185086661** [Dragon2fly/vpngate-with-proxy#18]
error picture:
http://wuguowei18508.com/Lychee/uploads/big/4c40997eb19f9f9095408064885a83a3.png

PS:I am a user from china,Does it matter?

**Comment by Dragon2fly** [comment]
Hi @wuhualu185086661 

The program tries to access http://www.vpngate.net/ . In case it failed, then it will try the next mirror link defined in `config.ini`. When all links return error, "Failed to get VPN servers data" shown up.

Be sure that you can access http://www.vpngate.net/ or any mirror in the link below
https://github.com/waylau/vpngate-mirrors
http://www.vpngate.net/en/sites.aspx

If you can't access any of those links, then The Great Firewall of China has blocked them all.
However, you can build a webserver to fetch http://www.vpngate.net/api/iphone/ file in any free host outside of China, encrypt and then download it.

**Comment by CarlosTheory** [comment]
Hi, I can not get connection neither. I'm getting a connection timeout when I execute ./run cli 

using gate:  http://www.vpngate.net
HTTPConnectionPool(host='www.vpngate.net', port=80): Read timed out.
Connection to gate http://www.vpngate.net failed

using gate:  http://p76ed4cd5.tokynt01.ap.so-net.ne.jp:16169
HTTPConnectionPool(host='p76ed4cd5.tokynt01.ap.so-net.ne.jp', port=16169): Max retries exceeded with url: /api/iphone/ (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7fdd72035250>, 'Connection to p76ed4cd5.tokynt01.ap.so-net.ne.jp timed out. (connect timeout=3)'))
Connection to gate http://p76ed4cd5.tokynt01.ap.so-net.ne.jp:16169 failed

using gate:  http://103.1.249.67:29858
HTTPConnectionPool(host='103.1.249.67', port=29858): Max retries exceeded with url: /api/iphone/ (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7fdd7208ea10>, 'Connection to 103.1.249.67 timed out. (connect timeout=3)'))
Connection to gate http://103.1.249.67:29858 failed

using gate:  http://211.217.242.42:3230
HTTPConnectionPool(host='211.217.242.42', port=3230): Max retries exceeded with url: /api/iphone/ (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7fdd7208eb10>, 'Connection to 211.217.242.42 timed out. (connect timeout=3)'))
Connection to gate http://211.217.242.42:3230 failed

using gate:  http://zp018093.ppp.dion.ne.jp:36205
HTTPConnectionPool(host='zp018093.ppp.dion.ne.jp', port=36205): Max retries exceeded with url: /api/iphone/ (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7fdd7208eb90>, 'Connection to zp018093.ppp.dion.ne.jp timed out. (connect timeout=3)'))
Connection to gate http://zp018093.ppp.dion.ne.jp:36205 failed

Failed to get VPN servers data
Check your network setting and proxy

**Comment by leewp14** [comment]
What if I can actually access the mirror sites on browser, but `./run cli` shows read timed out?

**Comment by Dragon2fly** [comment]
@leewp14 Is your browser using a proxy to connect to the mirror sites? 
Does it happen also with `./run tui`?

The `http_proxy` in the environment may affect the program.

**Comment by leewp14** [comment]
@Dragon2fly No, my browser has no proxy connection, and the `http_proxy` environment is not set. I'm also able to ping the mirror sites through terminal. 
But I retried with a normal wifi just now (previously I'm on mobile data hotspot) and it works. I'm not sure, maybe it is a problem with my data? (i already not using the mobile data anymore)
But anyway it works now, thanks!

**Comment by evadogstar** [comment]
@Dragon2fly I'm still at old previous python version. And have the same problem: I fight with this problem by adding more timeout seconds.
